interoperator/controllers/schedulers/sflabelselectorscheduler/sflabelselectorscheduler_controller.go

```diff
@@ -0,0 +1,245 @@
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/cluster/registry"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/constants"
+	"github.com/cloudfoundry-incubator/service-fabrik-broker/interoperator/pkg/errors"
+
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SFLabelSelectorScheduler reconciles a SFLabelSelectorScheduler object
+type SFLabelSelectorScheduler struct {
+	client.Client
+	Log             logr.Logger
+	scheme          *runtime.Scheme
+	clusterRegistry registry.ClusterRegistry
+}
+
+// Reconcile schedules the SFServiceInstance to one SFCluster and sets the ClusterID in
+// SFServiceInstance.Spec.ClusterID. It chooses the cluster with least number of
+// SFServiceInstances already deployed
+func (r *SFLabelSelectorScheduler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+	log := r.Log.WithValues("sfserviceinstance", req.NamespacedName)
+
+	instance := &osbv1alpha1.SFServiceInstance{}
+	err := r.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if apiErrors.IsNotFound(err) {
+			// Object not found, return.  Created objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	if instance.Spec.ClusterID == "" {
+		labelSelector, err := getLabelSelectorString(*instance, *r)
+		if err != nil || labelSelector == "" {
+			log.Info("Failed to get labelSelector string..", "error", err, "labelSelector", labelSelector)
+			return ctrl.Result{}, err
+		}
+
+		clusterID, err := r.schedule(labelSelector)
+		if err != nil {
+			log.Error(err, "Failed to schedule ", "labelSelector", labelSelector, "clusterID", clusterID)
+			return ctrl.Result{}, err
+		}
+
+		if clusterID != "" {
+			log.Info("setting clusterID", "clusterID", clusterID)
+			instance.Spec.ClusterID = clusterID
+			if err := r.Update(ctx, instance); err != nil {
+				log.Error(err, "failed to set cluster id", "clusterID", clusterID)
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func getLabelSelectorString(sfServiceInstance osbv1alpha1.SFServiceInstance, r SFLabelSelectorScheduler) (string, error) {
+	log := r.Log
+	ctx := context.Background()
+
+	sfNamespace := os.Getenv(constants.NamespaceEnvKey)
+	if sfNamespace == "" {
+		sfNamespace = constants.DefaultServiceFabrikNamespace
+	}
+	plan := &osbv1alpha1.SFPlan{}
+	namespacedName := types.NamespacedName{
+		Name:      sfServiceInstance.Spec.PlanID,
+		Namespace: sfNamespace,
+	}
+	err := r.Get(ctx, namespacedName, plan)
```

Comment:
Should'nt we add the request requeue logic here as well when we're unable to read the object ?

Reply:
It will be taken care [here](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/750/files/64930419dc44b9ff2ba8a2a376d1cdcb46bb8a62#diff-658fd3ff9a291fb4732b35398e5144e7R73)